services/kuber/server/domain/utils/autoscaler/cluster_autoscaler.go

```diff
@@ -147,7 +158,90 @@ func getK8sVersion(version string) (string, error) {
 		if minor < 25 {
 			return "v1.25.0", nil
 		}
-		return fmt.Sprintf("v%s.%s.%s", match[1], match[2], match[3]), nil
+		// Find latest autoscaler patch version
+		latestVersion, err := getLatestMinorVersion(version)
+		if err != nil {
+			log.Warn().Msg("Could not retrieve latest cluster-autoscaler version, fallback to defined Kubernetes version")
+			return fmt.Sprintf("v%s.%s.%s", match[1], match[2], match[3]), nil
+		}
+		return latestVersion, nil
 	}
 	return "", fmt.Errorf("failed to parse %s into autoscaler image tag [vX.Y.Z]", version)
 }
+
+// getLatestMinorVersion returns latest patch tag for given kubernetes version
+// Example: for v1.25.1 returns v1.25.3
+func getLatestMinorVersion(k8sVersion string) (string, error) {
+	var minor, patch int
+
+	tagList, err := getClusterAutoscaleVersions()
+	if err != nil {
+		return "", err
+	}
+
+	// Extract values from k8sVersion
+	_, err = fmt.Sscanf(k8sVersion, "v1.%d.%d", &minor, &patch)
+	if err != nil {
+		return "", err
+	}
+
+	// Find latest patch version
+	var latestPatch int
+	for _, semver := range tagList {
+		var listMinor, listPatch int
+		_, err := fmt.Sscanf(semver, "v1.%d.%d", &listMinor, &listPatch)
+		if err != nil {
+			return "", err
+		}
+		if minor == listMinor {
+			if listPatch > latestPatch {
+				latestPatch = listPatch
+			}
+		}
+	}
+
+	latestSemver := fmt.Sprintf("v1.%d.%d", minor, latestPatch)
+
+	return latestSemver, nil
+}
+
+// getClusterAutoscaleVersions query the CA registry to retrieve all available tags,
+// extracts proper semver from it, and returns a slice of available tags
+func getClusterAutoscaleVersions() ([]string, error) {
+	var TagsResponse TagsResponse
+
+	// Query CA registry
+	req, err := http.NewRequestWithContext(context.TODO(), http.MethodGet, clusterAutoscalerRegistry, nil)
+	if err != nil {
+		return nil, err
+	}
+	response, err := http.DefaultClient.Do(req)
+	if err != nil {
+		return nil, err
+	}
```

Comment:
This could by replaced by 

```go
resp, err := http.Get(clusterAutoscalerRegistry)
if err != nil {
...
}
...
```

Reply:
Initiali I did it like that but `golinter` was complaining 😢 
https://github.com/berops/claudie/commit/51994d4293f75bdddb53167751ace90eb0b554f7

Reply:
I see, linter seems too strict on this one.

We can keep this then.